admin-dev/themes/new-theme/js/pages/product/components/catalog-price-rule/catalog-price-rule-list.ts

```diff
@@ -0,0 +1,108 @@
+    const trTemplateContainer = this.listContainer.querySelector(CatalogPriceRuleMap.listRowTemplate) as HTMLScriptElement;
+    const rowContainer = document.querySelector(CatalogPriceRuleMap.catalogPriceRuleRow) as HTMLElement;
+    const editCatalogPriceRuleUrl = rowContainer.dataset.catalogPriceUrl;
+    if (!editCatalogPriceRuleUrl) {
+      return;
+    }
+
+    const trTemplate = trTemplateContainer.innerHTML as string;
+    tbody.innerHTML = '';
+
+    getCatalogRules().then((response) => {
+      const catalogPriceRules = response.catalogPriceRules as Array<CatalogPriceRuleForListing>;
+      this.toggleListVisibility(catalogPriceRules.length > 0);
+
+      catalogPriceRules.forEach((catalogPriceRule: CatalogPriceRuleForListing) => {
+        const temporaryContainer = document.createElement('tbody');
+        temporaryContainer.innerHTML = trTemplate.trim();
+
+        const trClone = temporaryContainer.firstChild as HTMLElement;
+        const idField = this.selectListField(trClone, listFields.catalogPriceRuleId);
+        const shopField = this.selectListField(trClone, listFields.shop);
+        const currencyField = this.selectListField(trClone, listFields.currency);
+        const countryField = this.selectListField(trClone, listFields.country);
+        const groupField = this.selectListField(trClone, listFields.group);
+        const nameField = this.selectListField(trClone, listFields.name);
+        const fromQuantityField = this.selectListField(trClone, listFields.fromQuantity);
+        const reductionTypeField = this.selectListField(trClone, listFields.reductionType);
+        const reductionField = this.selectListField(trClone, listFields.reduction);
+        const startDateField = this.selectListField(trClone, listFields.from);
+        const endDateField = this.selectListField(trClone, listFields.to);
+
+        const editBtn = this.selectLink(trClone, listFields.editBtn);
+        idField.textContent = String(catalogPriceRule.id);
+        shopField.textContent = catalogPriceRule.shop;
+        currencyField.textContent = catalogPriceRule.currency;
+        countryField.textContent = catalogPriceRule.country;
+        groupField.textContent = catalogPriceRule.group;
+        nameField.textContent = catalogPriceRule.name;
+        fromQuantityField.textContent = catalogPriceRule.fromQuantity;
+        reductionTypeField.textContent = catalogPriceRule.reductionType;
+        reductionField.textContent = catalogPriceRule.reduction;
+        startDateField.textContent = catalogPriceRule.startDate;
+        endDateField.textContent = catalogPriceRule.endDate;
+        editBtn.href = editCatalogPriceRuleUrl.replace('%catalog_price_rule_id%', String(catalogPriceRule.id));
+        tbody.append(trClone);
+      });
+    });
+  }
+
+  public toggleBlockVisibility(show: boolean): void {
+    this.listContainer.classList.toggle('d-none', !show);
+  }
```

Comment:
```suggestion
```

This one seems to be never used, I'm not sure we need to be able to hide the list after it's been shown. The button is only here initially because the list is hidden by default, but then the button should disappear and the list stays visible until the page is refreshed.